### PR TITLE
[13.0][ADD] l10n_nl_base_ubl

### PR DESCRIPTION
**Pull request by bosd**
Continuation of: https://github.com/OCA/l10n-netherlands/pull/270
Fixup the double commits

### CODE REVIEW

**Comment by bosd** [comment]
@thomaspaulb Please review this one as well 🙏 
@NL66278 Did the steps you outlined in the other PR. You like to review this one as well?

**Comment by thomaspaulb** [comment]
@bosd The code looks good to me at first glance.

Could you maybe test how well it plays together with [this PR](https://github.com/OCA/edi/pull/323/files) which is also ready for review? The goal there was to have fully compliant PEPPOL output that passes the validators.

**Comment by NL66278** [comment]
@bosd Maybe squash the fixup commits?

**Comment by bosd** [comment]
Done! Fixup Commits Squashed.

Now there is a merge conflict in the oca_dependencies.txt
Dunno how to fix that. :no_good: Hopefully someone can help

**Comment by gjotten** [comment]
> Done! Fixup Commits Squashed.
> 
> Now there is a merge conflict in the oca_dependencies.txt Dunno how to fix that. no_good Hopefully someone can help

@thomaspaulb or @NL66278 maybe one of you guys can help out @bosd with that ^^

**Comment by thomaspaulb** [comment]
The conflict seems to be solved, but Travis seems to be malfunctioning. Lemme try if I can deal with that,

**Comment by thomaspaulb** [comment]
[Awaiting review](https://github.com/OCA/l10n-netherlands/pull/365/files)

**Comment by thomaspaulb** [comment]
@bosd Can you rebase on latest 13.0

**Comment by bosd** [comment]
> Can you rebase on latest 13.0

@thomaspaulb  Done! Let's see how it goes!

**Review by thomaspaulb — APPROVED** [review]

**Comment by thomaspaulb** [comment]
@NL66278 Could you review

File: l10n_nl_base_ubl/tests/test_ubl_at_install.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            invoice_form.partner_id.country_id = self.env.ref("base.nl")
+            with invoice_form.invoice_line_ids.new() as invoice_line_form:
+                invoice_line_form.product_id = self.env.ref("product.product_product_4")
+                invoice_line_form.name = "product that cost 100"
+                invoice_line_form.quantity = 1
+                invoice_line_form.price_unit = 100.0
+            self.invoice = invoice_form.save()
+
+    def test_01_ubl_emulate_kvk_found(self):
+        nsmap, ns = self.env["base.ubl"]._ubl_get_nsmap_namespace("Invoice-1")
```
Comment (NL66278) [inline l10n_nl_base_ubl/tests/test_ubl_at_install.py:26]: I see no asserts in this test. Is this just checking that methods do not error out?

File: l10n_nl_base_ubl/tests/test_ubl_at_install.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        classBaseUbl = path_addon + path_file + "BaseUbl."
+
+        with patch(classBaseUbl + "_l10n_nl_base_ubl_get_kvk") as _sugg_mock:
+            _sugg_mock.return_value = "12345"
+
+            self.env["base.ubl"]._ubl_add_customer_party(
+                self.invoice.partner_id, False, "AccountingCustomerParty", xml_root, ns
+            )
+
+    def test_02_ubl_emulate_oin_found(self):
```
Comment (NL66278) [inline l10n_nl_base_ubl/tests/test_ubl_at_install.py:40]: Same here

File: l10n_nl_base_ubl/tests/test_ubl_at_install.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+from mock import patch
+
+from odoo.tests.common import Form, HttpCase
+
+
+class TestUblInvoice(HttpCase):
```
Comment (NL66278) [inline l10n_nl_base_ubl/tests/test_ubl_at_install.py]: Why HttpCase?

File: l10n_nl_base_ubl/tests/test_ubl_post_install.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+from lxml import etree
+
+import odoo
+from odoo.tests.common import Form, HttpCase
+
+
+@odoo.tests.tagged("post_install", "-at_install")
+class TestUblInvoice(HttpCase):
+    def setUp(self):
```
Comment (NL66278) [inline l10n_nl_base_ubl/tests/test_ubl_post_install.py:12]: Same setup as far as I can see in the other class. Maybe refactor todo the setUp in a common parent class?

File: l10n_nl_base_ubl/models/base_ubl.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+from lxml import etree
+
+from odoo import api, models
+
+
+class BaseUbl(models.AbstractModel):
+    _inherit = "base.ubl"
+
+    @api.model
+    def _ubl_add_party_legal_entity(self, partner, parent_node, ns, version="2.1"):
```
Comment (NL66278) [inline l10n_nl_base_ubl/models/base_ubl.py:13]: I s it not possible to add a test for this method?

File: l10n_nl_base_ubl/tests/test_ubl_at_install.py (hunk 3 above)
Comment (bosd) [reply l10n_nl_base_ubl/tests/test_ubl_at_install.py]: I don't know. Changed it to TransactionCase

**Review by NL66278 — APPROVED** [review]
:+1: LGTM. Had some remarks but should not be blocking.

**Pull request merged**